src/state.rs

```diff
@@ -77,36 +77,37 @@ impl BarState {
 
     pub(crate) fn update(&mut self, now: Instant, f: impl FnOnce(&mut ProgressState)) {
         f(&mut self.state);
-        self.tick(now);
```

Comment:
This seems like a big change in behavior? Why do this?

Reply:
My thinking is that the whole reason the user is calling `update` is because they want to do something specific with `ProgressState` that isn't already covered by the existing API. I can't imagine they'd want to do something very custom + tick. Though I agree it is a change in behavior, so I could back out this change if you want.

Reply:
It doesn't seem like a big deal to me to tick in this particular case? It's not like ticking is expensive, we currently tick for most other kinds of updates IIRC.

Reply:
Sounds good, I'll back that change out after dinner tonight.

Reply:
So I was expecting that you'd change everything back to tick, like before, but you only backed this particular change out. I'm still not sure that makes sense?